src/main/java/com/philips/swcoe/cerberus/hounds/JavaCodeMetricsWithDiff.java

```diff
@@ -61,15 +61,23 @@ public Integer call() throws Exception {
             this.validateFilePath(methodConfigFile);
             methodConfig = Files.readAllLines(new File(methodConfigFile).toPath(), Charset.defaultCharset());
         }
+        if("md".equalsIgnoreCase(reportFormat) || "html".equalsIgnoreCase(reportFormat)) {
+            this.validateReportFormat(metricFormat);
+        }
         CodeMetricsDiffService codeMetricsDiffService = new CodeMetricsDiffService(pathToPrevious, pathToCurrent);
         List<CodeMetricsClassResult> metricsResult = codeMetricsDiffService.getMetricsFromSourceCode();
-        char delimiter =  DelimeterForReport.valueOf(reportFormat.toUpperCase()).getDelimeter();
         if("vertical".equalsIgnoreCase(metricFormat)) {
-            this.writeToUI(new CodeMetricsVerticalWriterService(classConfig, methodConfig, delimiter).generateMetricsReport(metricsResult));
+            this.writeToUI(new CodeMetricsVerticalWriterService(classConfig, methodConfig, reportFormat.toUpperCase()).generateMetricsReport(metricsResult));
         } else {
-            this.writeToUI(new CodeMetricsHorizontalWriterService(classConfig, methodConfig, delimiter).generateMetricsReport(metricsResult));
+            this.writeToUI(new CodeMetricsHorizontalWriterService(classConfig, methodConfig, reportFormat.toUpperCase()).generateMetricsReport(metricsResult));
```

Comment:
Updates (like this commit) are becoming expensive because each involves editing all near-duplicate pairs of logic for method and class level metrics. This cost will only increase as updates increase and the cost to refactor later will also increase. Consider refactoring for arbitrary grouping of metrics (class, method, whatever) and express only the small variations in the code instead. They seem ~90% similar.

Reply:
You are correct @simaos , This should be last, I have abstracted out Metrics generator and Metrics Reporter, and now that abstraction process is complete. From now on any new format of report we can just add a new reporter.  The new reporter is a factory and has behaviours to generate report in the intended format .